### DISCUSSION TITLE
Some design questions about wiring / trying to adapt for own project

### DISCUSSION THREAD

**Discussion by WennySoft** [Lenti84/VdMot_Controller#92] [Q&A] [answered]
Hi,

first a big thanks for sharing this project! I am currently in need to drive some homematic HmIP motors and I hope I may take some ideas out of this project (personal project). As I have a slightly different setup I am not able to take this project directly and therefore am trying to adapt it. 

There are a few points I am wondering about:

- in wiring diagram there is a L293D used as motor driver but pins does not match (L293D has 16 pin and there are 20 pin used). According to image there is an XL293-20 XINLUDA used. What is the reason for the switch?
- why is there a multiplexing between OUTN1 and OUTN2 - separate driving should already be possible by OUTP0 to OUTP11 ?
- Ref2 is created twice - once from R50 on the current measurement and once by R54/R55. What is the purpose of these two ways? Are they both connected to the µp or are they alternatives?

Regards Michael

**Comment by Lenti84** [comment]
Hi, 
thanks for your interrest.
- L293D: seems to be a failure in the schematic symbol. Used XL293-20 as it is drop-in replacement and available.
- Multiplexing: With OUTN1 and OUTN2 multiplexing we can use 3x L293 to drive 12 motors. Without it would have been only 6 motors.
- Ref2 is also measured by STM32, as I wanted to catch possible differences between 3V3 supply and STM32 Reference voltage on BlackPill board. Ref2 is created once, but used twice. U4D is an impedance buffer.

Hope this helps.
Kind regards Andreas

**Comment by WennySoft** [comment]
Hi Andreas,
thanks for some clarification. I am missing something regarding the the impedance buffer. I understood that all 3 "Ref2" pins are wired together. But which is the creation side of Ref2?
At the right side of R50 or at the output of U4D? If R50 I would suggest that this is used as positive input of U4D. If U4D I do not understand why it is connected to R50?
How does this work?
Regards Michael

**Comment by Lenti84** [comment, marked as answer]
All Ref2 Pins are tied together.
R54/R55 is only a voltage devider to get half voltage of 3V3. U4D is buffering this voltage.
U12A is used as differential amplifier with offset by Ref2 voltage. Offset is feed in by R50.
STM32 measures both Ref2 and Current to evaluate the differential of both.

Regards

**Reply by WennySoft** [reply]
thanks - I got it.
Regards Michael
